### PR TITLE
server/cluster: migrate test framework to testify

### PR DESCRIPTION
**Pull request by LLThomas**
Signed-off-by: LLThomas <zs033@qq.com>

### What problem does this PR solve?

Issue Number: Ref #4813

### What is changed and how does it work?

```commit-message
As the title says.
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

**Comment by LLThomas** [comment]
coordinator_test.go and cluster_test.go have the same function named "checkRegion". What name should I change?

File: server/cluster/cluster_test.go [hunk 1]
```diff
@@ -85,30 +69,31 @@ func (s *testClusterInfoSuite) TestStoreHeartbeat(c *C) {
...
+		re.Equal(int64(0), store.GetLastHeartbeatTS().UnixNano())
 
-		c.Assert(cluster.HandleStoreHeartbeat(storeStats), IsNil)
+		re.NoError(cluster.HandleStoreHeartbeat(storeStats))
 
 		s := cluster.GetStore(store.GetID())
-		c.Assert(s.GetLastHeartbeatTS().UnixNano(), Not(Equals), int64(0))
-		c.Assert(s.GetStoreStats(), DeepEquals, storeStats)
+		re.NotEqual(int64(0), s.GetLastHeartbeatTS().UnixNano())
+
```
Comment (JmPotato) [inline server/cluster/cluster_test.go:83]: Please remove this empty line.

File: server/cluster/cluster_test.go [hunk 2]
```diff
@@ -137,56 +122,60 @@ func (s *testClusterInfoSuite) TestStoreHeartbeat(c *C) {
 		},
 		PeerStats: []*pdpb.PeerStat{},
 	}
-	c.Assert(cluster.HandleStoreHeartbeat(hotHeartBeat), IsNil)
-	c.Assert(cluster.HandleStoreHeartbeat(hotHeartBeat), IsNil)
-	c.Assert(cluster.HandleStoreHeartbeat(hotHeartBeat), IsNil)
+	re.Nil(cluster.HandleStoreHeartbeat(hotHeartBeat))
+	re.Nil(cluster.HandleStoreHeartbeat(hotHeartBeat))
+	re.Nil(cluster.HandleStoreHeartbeat(hotHeartBeat))
```
Comment (JmPotato) [inline server/cluster/cluster_test.go:127]:
```suggestion
	re.NoError(cluster.HandleStoreHeartbeat(hotHeartBeat))
	re.NoError(cluster.HandleStoreHeartbeat(hotHeartBeat))
	re.NoError(cluster.HandleStoreHeartbeat(hotHeartBeat))
```

File: server/cluster/cluster_test.go [hunk 3]
```diff
@@ -137,56 +122,60 @@ func (s *testClusterInfoSuite) TestStoreHeartbeat(c *C) {
...
-	c.Assert(storeStats[1][0].Loads[statistics.RegionReadBytes], Equals, float64(hotHeartBeat.PeerStats[0].ReadBytes)/interval)
-	c.Assert(storeStats[1][0].Loads[statistics.RegionReadKeys], Equals, float64(hotHeartBeat.PeerStats[0].ReadKeys)/interval)
-	c.Assert(storeStats[1][0].Loads[statistics.RegionReadQuery], Equals, float64(hotHeartBeat.PeerStats[0].QueryStats.Get)/interval)
+	re.Len(storeStats[1][0].Loads, int(statistics.RegionStatCount))
+	re.Equal(float64(hotHeartBeat.PeerStats[0].ReadBytes)/interval, storeStats[1][0].Loads[statistics.RegionReadBytes])
+	re.Equal(float64(hotHeartBeat.PeerStats[0].ReadKeys)/interval, storeStats[1][0].Loads[statistics.RegionReadKeys])
+	re.Equal(float64(hotHeartBeat.PeerStats[0].QueryStats.Get)/interval, storeStats[1][0].Loads[statistics.RegionReadQuery])
 	// After cold heartbeat, we won't find region 1 peer in regionStats
-	c.Assert(cluster.HandleStoreHeartbeat(coldHeartBeat), IsNil)
+	re.Nil(cluster.HandleStoreHeartbeat(coldHeartBeat))
```
Comment (JmPotato) [inline server/cluster/cluster_test.go:138]: Ditto.

File: server/cluster/cluster_test.go [hunk 4]
```diff
@@ -1022,87 +1079,95 @@ func (s *testClusterInfoSuite) TestRegionLabelIsolationLevel(c *C) {
...
-	checkRegion(c, cluster.GetRegionByKey([]byte("a")), region2)
+	re.NoError(cluster.processRegionHeartbeat(region1))
+	checkRegion(re, cluster.GetRegionByKey([]byte("z")), region1)
+	checkRegion(re, cluster.GetRegionByKey([]byte("a")), region2)
 	// [m, q) is missing before r3's heartbeat.
-	c.Assert(cluster.GetRegionByKey([]byte("n")), IsNil)
-	c.Assert(cluster.processRegionHeartbeat(region3), IsNil)
-	checkRegion(c, cluster.GetRegionByKey([]byte("n")), region3)
+	re.Nil(cluster.GetRegionByKey([]byte("n")))
+	re.Nil(cluster.processRegionHeartbeat(region3))
```
Comment (JmPotato) [inline server/cluster/cluster_test.go:1158]:
```suggestion
	re.NoError(cluster.processRegionHeartbeat(region3))
```

File: server/cluster/unsafe_recovery_controller_test.go [hunk 5]
```diff
@@ -183,18 +173,18 @@ func (s *testUnsafeRecoverySuite) TestFinished(c *C) {
...
 		// require peer report by empty plan
-		c.Assert(resp.RecoveryPlan, NotNil)
-		c.Assert(len(resp.RecoveryPlan.Creates), Equals, 0)
-		c.Assert(len(resp.RecoveryPlan.Demotes), Equals, 0)
-		c.Assert(resp.RecoveryPlan.ForceLeader, IsNil)
-		c.Assert(resp.RecoveryPlan.Step, Equals, uint64(1))
-		applyRecoveryPlan(c, storeID, reports, resp)
+		re.NotNil(resp.RecoveryPlan)
+		re.Equal(0, len(resp.RecoveryPlan.Creates))
+		re.Equal(0, len(resp.RecoveryPlan.Demotes))
```
Comment (lhy1024) [inline server/cluster/unsafe_recovery_controller_test.go:184]: how about `re.Empty`

File: server/cluster/unsafe_recovery_controller_test.go [hunk 6]
```diff
@@ -259,17 +253,17 @@ func (s *testUnsafeRecoverySuite) TestFailed(c *C) {
...
 		resp := &pdpb.StoreHeartbeatResponse{}
 		recoveryController.HandleStoreHeartbeat(req, resp)
-		c.Assert(resp.RecoveryPlan, NotNil)
-		c.Assert(len(resp.RecoveryPlan.Creates), Equals, 0)
-		c.Assert(len(resp.RecoveryPlan.Demotes), Equals, 0)
-		c.Assert(resp.RecoveryPlan.ForceLeader, IsNil)
-		applyRecoveryPlan(c, storeID, reports, resp)
+		re.NotNil(resp.RecoveryPlan)
+		re.Equal(0, len(resp.RecoveryPlan.Creates))
+		re.Equal(0, len(resp.RecoveryPlan.Demotes))
```
Comment (lhy1024) [inline server/cluster/unsafe_recovery_controller_test.go:264]: ditto

File: server/cluster/unsafe_recovery_controller_test.go [hunk 7]
```diff
@@ -552,29 +562,33 @@ func (s *testUnsafeRecoverySuite) TestOneLearner(c *C) {
 
 	for storeID, report := range reports {
 		if result, ok := expects[storeID]; ok {
-			c.Assert(report.PeerReports, DeepEquals, result.PeerReports)
+			re.Equal(result.PeerReports, report.PeerReports)
 		} else {
-			c.Assert(len(report.PeerReports), Equals, 0)
+			re.Equal(0, len(report.PeerReports))
```
Comment (lhy1024) [inline server/cluster/unsafe_recovery_controller_test.go:567]: ditto

File: server/cluster/unsafe_recovery_controller_test.go [hunk 8]
```diff
@@ -964,50 +986,58 @@ func (s *testUnsafeRecoverySuite) TestJointState(c *C) {
...
```
Comment (lhy1024) [inline server/cluster/unsafe_recovery_controller_test.go:991]: ditto

File: server/cluster/unsafe_recovery_controller_test.go [hunk 9]
```diff
@@ -1166,17 +1204,17 @@ func (s *testUnsafeRecoverySuite) TestOnHealthyRegions(c *C) {
...
 		resp := &pdpb.StoreHeartbeatResponse{}
 		recoveryController.HandleStoreHeartbeat(req, resp)
-		c.Assert(resp.RecoveryPlan, NotNil)
-		c.Assert(len(resp.RecoveryPlan.Creates), Equals, 0)
-		c.Assert(len(resp.RecoveryPlan.Demotes), Equals, 0)
-		c.Assert(resp.RecoveryPlan.ForceLeader, IsNil)
-		applyRecoveryPlan(c, storeID, reports, resp)
+		re.NotNil(resp.RecoveryPlan)
+		re.Equal(0, len(resp.RecoveryPlan.Creates))
+		re.Equal(0, len(resp.RecoveryPlan.Demotes))
```
Comment (lhy1024) [inline server/cluster/unsafe_recovery_controller_test.go:1215]: ditto

File: server/cluster/unsafe_recovery_controller_test.go [hunk 10]
```diff
@@ -1278,9 +1320,9 @@ func (s *testUnsafeRecoverySuite) TestCreateEmptyRegion(c *C) {
 
 	for storeID, report := range reports {
 		if expect, ok := expects[storeID]; ok {
-			c.Assert(report.PeerReports, DeepEquals, expect.PeerReports)
+			re.Equal(expect.PeerReports, report.PeerReports)
 		} else {
-			c.Assert(len(report.PeerReports), Equals, 0)
+			re.Equal(0, len(report.PeerReports))
```
Comment (lhy1024) [inline server/cluster/unsafe_recovery_controller_test.go:1325]: ditto

File: server/cluster/coordinator_test.go [hunk 11]
```diff
@@ -373,123 +369,129 @@ func (s *testCoordinatorSuite) TestCheckRegionWithScheduleDeny(c *C) {
...
 				op := newTestOperator(regionID, tc.GetRegion(regionID).GetRegionEpoch(), operatorKind)
-				c.Assert(co.opController.AddWaitingOperator(op), Equals, 1)
+				re.Equal(1, co.opController.AddWaitingOperator(op))
 			case operator.OpRegion | operator.OpMerge:
 				if regionID%2 == 1 {
 					ops, err := operator.CreateMergeRegionOperator("merge-region", co.cluster, tc.GetRegion(regionID), tc.GetRegion(regionID-1), operator.OpMerge)
-					c.Assert(err, IsNil)
-					c.Assert(co.opController.AddWaitingOperator(ops...), Equals, len(ops))
+					re.NoError(err)
+					re.Equal(len(ops), co.opController.AddWaitingOperator(ops...))
```
Comment (lhy1024) [inline server/cluster/coordinator_test.go:405]: how about `re.Len`

File: server/cluster/store_limiter_test.go [hunk 12]
```diff
@@ -15,43 +15,39 @@
 package cluster
 
 import (
-	. "github.com/pingcap/check"
 	"github.com/pingcap/kvproto/pkg/pdpb"
+	"github.com/stretchr/testify/require"
 	"github.com/tikv/pd/server/config"
 	"github.com/tikv/pd/server/core/storelimit"
+	"testing"
```
[the remark this answers is not in the document]
Comment (LLThomas) [reply server/cluster/store_limiter_test.go:22]: If I move "testing" before line 18, "make check" can not pass. - -

[the remark this answers is not in the document]
Comment (rleungx) [reply server/cluster/store_limiter_test.go:22]: Can you paste the error?

[the remark this answers is not in the document]
Comment (LLThomas) [reply server/cluster/store_limiter_test.go]: It was my mistake. I'll fix it.

File: server/cluster/cluster_test.go [hunk 13]
```diff
@@ -137,56 +121,60 @@ func (s *testClusterInfoSuite) TestStoreHeartbeat(c *C) {
...
 	// After cold heartbeat, we won't find region 1 peer in regionStats
-	c.Assert(cluster.HandleStoreHeartbeat(coldHeartBeat), IsNil)
+	re.NoError(cluster.HandleStoreHeartbeat(coldHeartBeat))
 	time.Sleep(20 * time.Millisecond)
 	storeStats = cluster.hotStat.RegionStats(statistics.Read, 1)
-	c.Assert(storeStats[1], HasLen, 0)
+	re.Len(storeStats[1], 0)
 	// After hot heartbeat, we can find region 1 peer again
-	c.Assert(cluster.HandleStoreHeartbeat(hotHeartBeat), IsNil)
+	re.Nil(cluster.HandleStoreHeartbeat(hotHeartBeat))
```
Comment (JmPotato) [inline server/cluster/cluster_test.go:142]:
```suggestion
	re.NoError(cluster.HandleStoreHeartbeat(hotHeartBeat))
```

File: server/cluster/cluster_test.go [hunk 14]
```diff
@@ -137,56 +121,60 @@ func (s *testClusterInfoSuite) TestStoreHeartbeat(c *C) {
...
+	re.Len(storeStats[1], 1)
+	re.Equal(uint64(1), storeStats[1][0].RegionID)
 	storeStats = cluster.hotStat.RegionStats(statistics.Read, 3)
-	c.Assert(storeStats[1], HasLen, 0)
+	re.Len(storeStats[1], 0)
 	// after 2 hot heartbeats, wo can find region 1 peer again
-	c.Assert(cluster.HandleStoreHeartbeat(hotHeartBeat), IsNil)
-	c.Assert(cluster.HandleStoreHeartbeat(hotHeartBeat), IsNil)
+	re.Nil(cluster.HandleStoreHeartbeat(hotHeartBeat))
+	re.Nil(cluster.HandleStoreHeartbeat(hotHeartBeat))
```
Comment (JmPotato) [inline server/cluster/cluster_test.go:163]: Ditto.

**Review by JmPotato — APPROVED** [review]
The rest LGTM.

File: server/cluster/unsafe_recovery_controller_test.go [hunk 15]
```diff
@@ -775,26 +793,30 @@ func (s *testUnsafeRecoverySuite) TestUninitializedPeer(c *C) {
 
 	for storeID, report := range reports {
 		if result, ok := expects[storeID]; ok {
-			c.Assert(report.PeerReports, DeepEquals, result.PeerReports)
+			re.Equal(result.PeerReports, report.PeerReports)
 		} else {
-			c.Assert(len(report.PeerReports), Equals, 0)
+			re.Equal(0, len(report.PeerReports))
```
Comment (rleungx) [inline server/cluster/unsafe_recovery_controller_test.go:798]:
```suggestion
			re.Empty(report.PeerReports)
```

File: server/cluster/unsafe_recovery_controller_test.go [hunk 16]
```diff
@@ -1471,72 +1521,80 @@ func (s *testUnsafeRecoverySuite) TestRangeOverlap2(c *C) {
...
```
Comment (rleungx) [inline server/cluster/unsafe_recovery_controller_test.go:1526]:
```suggestion
			re.Empty(report.PeerReports)
```

**Review by rleungx — APPROVED** [review]
The rest LGTM

**Pull request merged**